Signal/src/ViewControllers/MessageComposeTableViewController.m

```diff
@@ -101,6 +115,20 @@ - (void)loadView
     [self updateTableContents];
 }
 
+- (void)showContactsPermissionReminder:(BOOL)isVisible
+{
+    _hideContactsPermissionReminderViewConstraint.active = !isVisible;
+}
+
+- (void)showSearchBar:(BOOL)isVisible
+{
+    if (isVisible) {
+        self.tableViewController.tableView.tableHeaderView = self.searchBar;
+    } else {
+        self.tableViewController.tableView.tableHeaderView = nil;
```

Comment:
If we hide the search bar, users can't "invite non-contacts via SMS."  It might also be nice to modify `SelectRecipientViewController` to offer to "invite via SMS" when it's `shouldValidatePhoneNumbers` delegate property is true.